File: web-frontend/modules/integrations/ai/nodeTypes.js
import { NodeType } from '@baserow/modules/automation/nodeTypes'
import { ActionNodeTypeMixin } from '@baserow/modules/automation/nodeTypeMixins'
import { AIAgentServiceType } from '@baserow/modules/integrations/ai/serviceTypes'

export class AIAgentActionNodeType extends ActionNodeTypeMixin(NodeType) {
  static getType() {
    return 'ai_agent'
  }

  get name() {
    return this.app.i18n.t('nodeType.aiAgent')
  }

  get iconClass() {
    return 'iconoir-sparks'
  }

  get serviceType() {
    return this.app.$registry.get('service', AIAgentServiceType.getType())
  }

  getOrder() {
    return 8
  }
}

File: backend/src/baserow/contrib/integrations/ai/node_types.py
from baserow.contrib.automation.nodes.models import AIAgentActionNode
from baserow.contrib.automation.nodes.node_types import AutomationNodeActionNodeType
from baserow.contrib.integrations.ai.service_types import AIAgentServiceType


class AIAgentNodeType(AutomationNodeActionNodeType):
    type = "ai_agent"
    model_class = AIAgentActionNode
    service_type = AIAgentServiceType.type
